core/src/replay_stage.rs

```diff
@@ -3533,22 +3540,22 @@ impl ReplayStage {
 
             let propagation_confirmed = is_leader_slot
                 || progress
-                    .get_leader_propagation_slot_must_exist(bank.slot())
+                    .get_leader_propagation_slot_must_exist(check_slot)
                     .0;
 
             if is_locked_out {
-                failure_reasons.push(HeaviestForkFailures::LockedOut(bank.slot()));
+                failure_reasons.push(HeaviestForkFailures::LockedOut(check_slot));
```

Comment:
So do we have any tests check the slot in failure_reasons? Because I don't see any tests included in this PR.

Reply:
No old tests as far as I know, but now that https://github.com/solana-labs/solana/pull/33341 has landed I've rebased and removed the hack for the test in that PR.